Filtered_kernel/include/CGAL/internal/Static_filters/Compare_distance_3.h

```diff
@@ -0,0 +1,166 @@
+
+public:
+
+  typedef typename Base::result_type  result_type;
+
+  using Base::operator();
+
+  result_type operator()(const Point_3 &p, const Point_3& q, const Point_3& r) const
+  {
+    CGAL_BRANCH_PROFILER(std::string("semi-static attempts/calls to   : ") +
+                         std::string(CGAL_PRETTY_FUNCTION), tmp);
+
+    Get_approx<Point_3> get_approx; // Identity functor for all points
+                                    // but lazy points
+
+    if(q == r){
+      return EQUAL;
+    }
+    double px, py, pz, qx, qy, qz, rx, ry, rz;
+
+    if (fit_in_double(get_approx(p).x(), px) && fit_in_double(get_approx(p).y(), py) &&
+        fit_in_double(get_approx(p).z(), pz) &&
+        fit_in_double(get_approx(q).x(), qx) && fit_in_double(get_approx(q).y(), qy) &&
+        fit_in_double(get_approx(q).z(), qz) &&
+        fit_in_double(get_approx(r).x(), rx) && fit_in_double(get_approx(r).y(), ry) &&
+        fit_in_double(get_approx(r).z(), rz))
+    {
+      CGAL_BRANCH_PROFILER_BRANCH(tmp);
+      double qpx;
+      qpx = (qx - px);
+      double qpy;
+      qpy = (qy - py);
+      double qpz;
+      qpz = (qz - pz);
+      double qp2;
+      qp2 = ((square( qpx ) + square( qpy )) + square( qpz ));
+      double rpx;
+      rpx = (rx - px);
+      double rpy;
+      rpy = (ry - py);
+      double rpz;
+      rpz = (rz - pz);
+      double rp2;
+      rp2 = ((square( rpx ) + square( rpy )) + square( rpz ));
+      Sign int_tmp_result = EQUAL;
+      double double_tmp_result;
+      double eps;
+      double_tmp_result = (qp2 - rp2);
+      double max1 = CGAL::abs(qpx);
+      if( (max1 < CGAL::abs(qpy)) )
+        {
+          max1 = CGAL::abs(qpy);
+        }
+      if( (max1 < CGAL::abs(qpz)) )
+        {
+          max1 = CGAL::abs(qpz);
+        }
+      if( (max1 < CGAL::abs(rpx)) )
+        {
+          max1 = CGAL::abs(rpx);
+        }
+      if( (max1 < CGAL::abs(rpy)) )
+        {
+          max1 = CGAL::abs(rpy);
+        }
+      if( (max1 < CGAL::abs(rpz)) )
+        {
+          max1 = CGAL::abs(rpz);
+        }
+      if( (max1 < 2.42701102401884262260e-147) )
+        {
+          return Base::operator()(p, q, r);
+        }
+      else
+        {
+          if( (max1 > 8.37987995621411946582e+152) )
+            {
+              return Base::operator()(p, q, r);
+            }
+          eps = (3.77746921267322435884e-15 * (max1 * max1));
+          if( (double_tmp_result > eps) )
+            {
+              int_tmp_result = LARGER;
+            }
+          else
+            {
+              if( (double_tmp_result < -eps) )
+                {
+                  int_tmp_result = SMALLER;
+                }
+              else
+                {
+                  return Base::operator()(p, q, r);
+                }
+            }
+        }
+      return int_tmp_result;
+    }
+    return Base::operator()(p, q, r);
+  }
+
+
+  result_type operator()(const Vector_3 &p, const Vector_3& q) const
```

Comment:
What is this one about?

Reply:
A copy-paste error.   I started from the file `Equal_3.h`.  Thanks for pointing it out.